sourcegraph/jetbrains
# Add a text field for entering multiple remote repositories at once

This facilitates bulk editing of multiple remote repositories, with an autocompleter and fuzzy search to make it easy to find repositories.

This needs to land after sourcegraph/cody#3775 and roll past that Agent version.

Fixes #982 

## Test plan

Manually tested with sourcegraph/cody#3775:

- Switch from an Enterprise to dotcom account and back. The enhanced context selector in the chat panel should change presentation.
- Click the pencil and enter repository names. Duplicates should have yellow underlines. More than 10 repositories should have yellow underlines. Unknown repositories should have red underlines. Note, in JetBrains 2022.x, Annotators are not DumbAware: The underlines will only appear after the project finishes indexing.
- Ctrl-Space should open an autocomplete with fuzzy search of remote repositories.
- Command/Ctrl-Enter should update the selected repositories in the enhanced context selector. The entered text, including errors, should persist for the life of the chat panel to facilitate incremental editing; new chat panels should pick up a list of repositories separated with newlines.
- Opening and closing the enhanced context selector tree view should resize the bottom of the panel. Checking and unchecking the tree nodes should influence whether enhanced context is used for chats (root node) or for specific repositories (leaf nodes.) Note, the repository associated with the open workspace is included implicitly when enhanced context is enabled.
